coreneuron/gpu/nrn_acc_manager.cpp

```diff
@@ -43,56 +43,49 @@ void nrn_ion_global_map_delete_from_device();
 void nrn_VecPlay_copyto_device(NrnThread* nt, void** d_vecplay);
 void nrn_VecPlay_delete_from_device(NrnThread* nt);
 
-void* cnrn_gpu_copyin(void* h_ptr, std::size_t len) {
+void* cnrn_target_deviceptr(void* h_ptr) {
```

Comment:
Maybe it would be more consistent to make a template too?
```c++
template <typename T>
T* cnrn_target_deviceptr(T* h_ptr) {
```
and I also think we should be more consistent with the naming.
In the current draft we seem to use 1-2 of `device`, `gpu` and `target` at random 🙃 
- `cnrn_target_deviceptr`
- `cnrn_gpu_copyin`
- `cnrn_target_delete`
- `cnrn_memcpy_to_device`

I don't mind too much as long as they're consistent. For example, the endings could be
- `deviceptr`
- `copyin`
- `delete`
- `memcpy_to_device`

following OpenACC, and the prefix could be `cnrn_target_` or `coreneuron::target::`?

Reply:
`cnrn_target` prefix looks good to me as well. Also the same for the rest of the functions